test/src/test/java/org/corfudb/infrastructure/logreplication/LogReplicationFSMTest.java

```diff
@@ -253,7 +253,9 @@ private void testSnapshotSender(int batchSize) throws Exception {
         transitionAvailable.acquire();
 
         // Transition #1: Snapshot Sync Request
-        UUID snapshotSyncRequestId = transition(LogReplicationEventType.SNAPSHOT_SYNC_REQUEST, LogReplicationStateType.IN_SNAPSHOT_SYNC);
+        LogReplicationEvent event = new LogReplicationEvent(LogReplicationEventType.SNAPSHOT_SYNC_REQUEST);
+        UUID snapshotSyncRequestId = event.getEventId();
+        fsm.input(event);
```

Comment:
Does fsm.input would take care of the transition?

Reply:
Yes in this test class we emulate the state transition (by inputting events directly to fsm). Here the check is for when we input a SNAPSHOT_SYNC_REQUEST, verify the snapshot sync actually happens (we wait for WAIT_SNAPSHOT_APPLY state below).

Before this change we were also checking IN_SNAPSHOT_SYNC state, but that state could be already transited into WAIT_SNAPSHOT_APPLY state at the check point (which is automatically handled by fsm)